spec/rubocop/rspec/hook_spec.rb

```diff
@@ -11,43 +11,89 @@ def hook(source)
-  end
+    it 'knows the scope of a hook with a symbol literal' do
+      expect(hook('before(:example) { example_setup }').knowable_scope?)
+        .to be(true)
+    end
 
-  it 'ignores other arguments to hooks' do
-    expect(hook('before(:each, :metadata) { example_setup }').scope)
-      .to be(:each)
-  end
+    it 'knows the scope of a hook with no argument' do
+      expect(hook('before { example_setup }').knowable_scope?)
+        .to be(true)
+    end
 
-  it 'classifies nonstandard hook arguments as invalid' do
-    expect(hook('before(:nothing) { example_setup }').valid_scope?).to be(false)
+    it 'knows the scope of a hook with hash metadata' do
+      expect(hook('before(special: true) { example_setup }').knowable_scope?)
+        .to be(true)
+    end
   end
 
-  it 'classifies :each as a valid hook argument' do
-    expect(hook('before(:each) { example_setup }').valid_scope?).to be(true)
-  end
+  describe '#scope' do
+    it 'ignores other arguments to hooks' do
+      expect(hook('before(:each, :metadata) { example_setup }').scope)
+        .to be(:each)
+    end
 
-  it 'classifies :each as an example hook' do
-    expect(hook('before(:each) { }').example?).to be(true)
-  end
+    it 'classifies :each as an example hook' do
+      expect(hook('before(:each) { }').example?).to be(true)
+    end
+
+    it 'defaults to example hook with hash metadata' do
+      expect(hook('before(special: true) { }').example?).to be(true)
+    end
 
-  shared_examples 'standardizes scope' do |source, scope|
-    it "interprets #{source} as having scope #{scope}" do
-      expect(hook(source).scope).to equal(scope)
+    shared_examples 'standardizes scope' do |source, scope|
+      it "interprets #{source} as having scope #{scope}" do
+        expect(hook(source).scope).to equal(scope)
+      end
     end
+
+    include_examples 'standardizes scope', 'before(:each) { }',    :each
+    include_examples 'standardizes scope', 'around(:example) { }', :each
+    include_examples 'standardizes scope', 'after { }',            :each
+
+    include_examples 'standardizes scope', 'before(:all) { }',     :context
+    include_examples 'standardizes scope', 'around(:context) { }', :context
+
+    include_examples 'standardizes scope', 'after(:suite) { }', :suite
   end
 
-  include_examples 'standardizes scope', 'before(:each) { }',    :each
-  include_examples 'standardizes scope', 'around(:example) { }', :each
-  include_examples 'standardizes scope', 'after { }',            :each
+  describe '#metadata' do
+    def metadata(source)
+      hook(source).metadata.to_a
+    end
 
-  include_examples 'standardizes scope', 'before(:all) { }',     :context
-  include_examples 'standardizes scope', 'around(:context) { }', :context
+    it 'extracts symbol metadata' do
+      expect(metadata('before(:example, :special) { foo }'))
+        .to contain_exactly([have_attributes(value: :special), true])
```

Comment:
Reading the `have_attributes(value: :special)` matcher, I’m not sure what’s actually in the metadata array.

Reply:
`metadata('before(:example, :special) { foo }')` (`metadata('before(:example, :special) { foo }').metadata.to_a`)is `[[s(:sym, :special), true]]`.

Do you think something like that in mind that would make it more indicative?
```ruby
    def metadata(source)
      hook(source).metadata
    end

    def a_sex_symbol(name)
      be_sym_type
        .and have_attributes(value: name)
    end

it do
  expect(metadata(...))
    .to eq(a_sex_symbol(:special) => true)
end
```

Unfortunately, I couldn't make it work with `eq`/`match`/`include`, they don't seem to match hash keys against argument matchers.

Reply:
Would it be too obscure to just call `#to_s` on the metadata?
```rb
def metadata(source)
  hook(source).metadata.to_s
end

it 'extracts symbol metadata' do
  expect(metadata('before(:example, :special) { foo }'))
    .to eq('{s(:sym, :special)=>true}')
end
```

Reply:
Not too bad. We can think later how to improve it if it's not clear enough.